### PR TITLE
Fix sanitizers

### LINKED ISSUE #64086: TSan/MSan builds failures
### TSan

```
May 17 18:34:12 FAILED: contrib/arrow-cmake/orc_proto.pb.h contrib/arrow-cmake/orc_proto.pb.cc /build/build_docker/contrib/arrow-cmake/orc_proto.pb.h /build/build_docker/contrib/arrow-cmake/orc_proto.pb.cc 
May 17 18:34:12 cd /build/build_docker/contrib/arrow-cmake && /build/build_docker/contrib/google-protobuf-cmake/protoc -I /build/contrib/orc/c++/../proto --cpp_out="/build/build_docker/contrib/arrow-cmake" /build/contrib/orc/c++/../proto/orc_proto.proto
May 17 18:34:12 ThreadSanitizer: CHECK failed: tsan_platform_linux.cpp:282 "((personality(old_personality | ADDR_NO_RANDOMIZE))) != ((-1))" (0xffffffffffffffff, 0xffffffffffffffff) (tid=123304)
May 17 18:34:12 Segmentation fault (core dumped)
```

### MSan

```
May 17 18:31:34 [3061/12202] Generating orc_proto.pb.h, orc_proto.pb.cc
May 17 18:31:34 FAILED: contrib/arrow-cmake/orc_proto.pb.h contrib/arrow-cmake/orc_proto.pb.cc /build/build_docker/contrib/arrow-cmake/orc_proto.pb.h /build/build_docker/contrib/arrow-cmake/orc_proto.pb.cc 
May 17 18:31:34 cd /build/build_docker/contrib/arrow-cmake && /build/build_docker/contrib/google-protobuf-cmake/protoc -I /build/contrib/orc/c++/../proto --cpp_out="/build/build_docker/contrib/arrow-cmake" /build/contrib/orc/c++/../proto/orc_proto.proto
May 17 18:31:34 FATAL: Code 0x603049277240 is out of application range. Non-PIE build?
May 17 18:31:34 FATAL: MemorySanitizer can not mmap the shadow memory.
May 17 18:31:34 FATAL: Make sure to compile with -fPIE and to link with -pie.
May 17 18:31:34 FATAL: Disabling ASLR is known to cause this error.
May 17 18:31:34 FATAL: If running under GDB, try 'set disable-randomization off'.
May 17 18:31:34 ==110628==Process memory map follows:
May 17 18:31:34 	0x603049036000-0x60304925a000	/build/build_docker/contrib/google-protobuf-cmake/protoc
May 17 18:31:34 	0x60304925a000-0x60304a10a000	/build/build_docker/contrib/google-protobuf-cmake/protoc
May 17 18:31:34 	0x60304a10a000-0x60304a121000	/build/build_docker/contrib/google-protobuf-cmake/protoc
May 17 18:31:34 	0x60304a121000-0x60304a128000	/build/build_docker/contrib/google-protobuf-cmake/protoc
May 17 18:31:34 	0x60304a128000-0x60304bb14000	
May 17 18:31:34 	0x7e0b4d700000-0x7e0b4d800000	
May 17 18:31:34 	0x7e0b4d900000-0x7e0b4da00000	
May 17 18:31:34 	0x7e0b4db00000-0x7e0b4dc00000	
May 17 18:31:34 	0x7e0b4dd00000-0x7e0b4de00000	
May 17 18:31:34 	0x7e0b4dee3000-0x7e0b4e291000	
May 17 18:31:34 	0x7e0b4e291000-0x7e0b4e292000	/usr/lib/x86_64-linux-gnu/libdl.so.2
May 17 18:31:34 	0x7e0b4e292000-0x7e0b4e293000	/usr/lib/x86_64-linux-gnu/libdl.so.2
May 17 18:31:34 	0x7e0b4e293000-0x7e0b4e294000	/usr/lib/x86_64-linux-gnu/libdl.so.2
May 17 18:31:34 	0x7e0b4e294000-0x7e0b4e295000	/usr/lib/x86_64-linux-gnu/libdl.so.2
May 17 18:31:34 	0x7e0b4e295000-0x7e0b4e296000	/usr/lib/x86_64-linux-gnu/libdl.so.2
May 17 18:31:34 	0x7e0b4e296000-0x7e0b4e297000	/usr/lib/x86_64-linux-gnu/librt.so.1
May 17 18:31:34 	0x7e0b4e297000-0x7e0b4e298000	/usr/lib/x86_64-linux-gnu/librt.so.1
May 17 18:31:34 	0x7e0b4e298000-0x7e0b4e299000	/usr/lib/x86_64-linux-gnu/librt.so.1
May 17 18:31:34 	0x7e0b4e299000-0x7e0b4e29a000	/usr/lib/x86_64-linux-gnu/librt.so.1
May 17 18:31:34 	0x7e0b4e29a000-0x7e0b4e29b000	/usr/lib/x86_64-linux-gnu/librt.so.1
May 17 18:31:34 	0x7e0b4e29b000-0x7e0b4e2a9000	/usr/lib/x86_64-linux-gnu/libm.so.6
May 17 18:31:34 	0x7e0b4e2a9000-0x7e0b4e325000	/usr/lib/x86_64-linux-gnu/libm.so.6
May 17 18:31:34 	0x7e0b4e325000-0x7e0b4e380000	/usr/lib/x86_64-linux-gnu/libm.so.6
May 17 18:31:34 	0x7e0b4e380000-0x7e0b4e381000	/usr/lib/x86_64-linux-gnu/libm.so.6
May 17 18:31:34 	0x7e0b4e381000-0x7e0b4e382000	/usr/lib/x86_64-linux-gnu/libm.so.6
May 17 18:31:34 	0x7e0b4e382000-0x7e0b4e3aa000	/usr/lib/x86_64-linux-gnu/libc.so.6
May 17 18:31:34 	0x7e0b4e3aa000-0x7e0b4e53f000	/usr/lib/x86_64-linux-gnu/libc.so.6
May 17 18:31:34 	0x7e0b4e53f000-0x7e0b4e597000	/usr/lib/x86_64-linux-gnu/libc.so.6
May 17 18:31:34 	0x7e0b4e597000-0x7e0b4e598000	/usr/lib/x86_64-linux-gnu/libc.so.6
May 17 18:31:34 	0x7e0b4e598000-0x7e0b4e59c000	/usr/lib/x86_64-linux-gnu/libc.so.6
May 17 18:31:34 	0x7e0b4e59c000-0x7e0b4e59e000	/usr/lib/x86_64-linux-gnu/libc.so.6
May 17 18:31:34 	0x7e0b4e59e000-0x7e0b4e5ab000	
May 17 18:31:34 	0x7e0b4e5ab000-0x7e0b4e5ac000	/usr/lib/x86_64-linux-gnu/libpthread.so.0
May 17 18:31:34 	0x7e0b4e5ac000-0x7e0b4e5ad000	/usr/lib/x86_64-linux-gnu/libpthread.so.0
May 17 18:31:34 	0x7e0b4e5ad000-0x7e0b4e5ae000	/usr/lib/x86_64-linux-gnu/libpthread.so.0
May 17 18:31:34 	0x7e0b4e5ae000-0x7e0b4e5af000	/usr/lib/x86_64-linux-gnu/libpthread.so.0
May 17 18:31:34 	0x7e0b4e5af000-0x7e0b4e5b0000	/usr/lib/x86_64-linux-gnu/libpthread.so.0
May 17 18:31:34 	0x7e0b4e5b1000-0x7e0b4e5b7000	
May 17 18:31:34 	0x7e0b4e5b7000-0x7e0b4e5b9000	/usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
May 17 18:31:34 	0x7e0b4e5b9000-0x7e0b4e5e3000	/usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
May 17 18:31:34 	0x7e0b4e5e3000-0x7e0b4e5ee000	/usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
May 17 18:31:34 	0x7e0b4e5ee000-0x7e0b4e5ef000	
May 17 18:31:34 	0x7e0b4e5ef000-0x7e0b4e5f1000	/usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
May 17 18:31:34 	0x7e0b4e5f1000-0x7e0b4e5f3000	/usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
May 17 18:31:34 	0x7ffd73396000-0x7ffd733b7000	[stack]
May 17 18:31:34 	0x7ffd733c1000-0x7ffd733c5000	[vvar]
May 17 18:31:34 	0x7ffd733c5000-0x7ffd733c7000	[vdso]
May 17 18:31:34 	0xffffffffff600000-0xffffffffff601000	[vsyscall]
May 17 18:31:34 ==110628==End of process memory map.
```

It looks like ASLR related, though not sure it pops up only now, maybe there was some changes on CI?

CI: https://s3.amazonaws.com/clickhouse-test-reports/64058/5ad98e88a254c6c9bc67af52b43702356d6c3950/clickhouse_build_check/report.html
Play: https://play.clickhouse.com/play?user=play#U0VMRUNUIGNoZWNrX3N0YXJ0X3RpbWUsIGNoZWNrX3N0YXR1cywgcmVwb3J0X3VybApGUk9NIGNoZWNrcwpXSEVSRSByZXBvcnRfdXJsIExJS0UgJyVjbGlja2hvdXNlX2J1aWxkX2NoZWNrJScKICAgIEFORCBjaGVja19zdGFydF90aW1lID49IG5vdygpIC0gSU5URVJWQUwgMyBEQVkKICAgIGFuZCBwdWxsX3JlcXVlc3RfbnVtYmVyID0gMAogICAgQU5EIGNoZWNrX3N0YXR1cyBub3QgaW4gKCdzdWNjZXNzJywgJ3BlbmRpbmcnKQpPUkRFUiBCWSBjaGVja19zdGFydF90aW1l

### PR DESCRIPTION
**Pull request by azat**
Since likely the image will not be updated from CI, on other words it doesn't looks like the part that I could run by myself, I've changed couple of other places to test will this helps or not within PR.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Fixes: https://github.com/ClickHouse/ClickHouse/issues/64086
Cc: @Felixoid (if this will work, can you please update the image)

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 990f7e43388393a52bfce6c028a779dcf126b74f with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64090/990f7e43388393a52bfce6c028a779dcf126b74f/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by azat** [comment]
Another try - https://github.com/ClickHouse/ClickHouse/pull/64091

**Comment by azat** [comment]
Interesting, builds passed, but stateless tests failed

**Review by alexey-milovidov — APPROVED** [review]
I'm not sure what needs to be done to rebuild AMI.
Maybe this will help after these steps.

**Comment by azat** [comment]
Actually I don't see that it helps, at least not 100%, so I'm still looking for a solution

**Comment by azat** [comment]
Interesting, looks like it helped

File: docker/packager/packager [hunk 1]
```diff
@@ -17,6 +17,17 @@ class BuildException(Exception):
     pass
 
 
+def apply_workarounds():
+    # workaround for sanitizers and some kernels [1].
+    #
+    #  [1]: https://github.com/google/sanitizers/issues/856
+    subprocess.check_call("sudo sysctl vm.mmap_rnd_bits", shell=True)
+    subprocess.check_call("sudo sysctl vm.mmap_rnd_bits=28", shell=True)
```
Comment (azat) [inline docker/packager/packager:25]: This is likely not required anymore, turning off ASLR should be enough I guess

**Comment by alesapin** [comment]
It's strange that some tests fail with weird error:

image

**Comment by Felixoid** [comment]
Here are all the changes made to the AMI on Friday

```
> git diff --word-diff /tmp/sysctl.{old,new} | grep '+}' | grep vm.
vm.min_free_kbytes = [-45056-]{+67584+}
vm.mmap_rnd_bits = [-28-]{+32+}
vm.mmap_rnd_compat_bits = [-8-]{+16+}
vm.user_reserve_kbytes = [-52114-]{+131072+}
```

I'm sorry about that. The update was made based on the hard deprecation of the actions runner.  This deprecation will be effective this coming Thursday.

If it helps, we can deploy a new AMI tomorrow. There's another tiny fix to deploy.

**Comment by alexey-milovidov** [comment]
> vm.mmap_rnd_bits = [-28-]{+32+}
vm.mmap_rnd_compat_bits = [-8-]{+16+}

Yes, this is exactly the reason.

> I'm sorry about that. The update was made based on the hard deprecation of the actions runner. This deprecation will be effective this coming Thursday.

No problem; maybe more visibility of AMI updates is needed.

**Pull request merged**